pkg/ipam-controller/allocator/allocator.go

```diff
@@ -90,37 +90,41 @@ func (pa *poolAllocator) Allocate(ctx context.Context, node string) (nodeAllocat
 	}
 	allocations := pa.getAllocationsAsSlice()
 	var startIP net.IP
-	if len(allocations) == 0 || ip.Distance(ip.Network(pa.cfg.Subnet).IP, allocations[0].StartIP) > 2 {
+	if len(allocations) == 0 || ip.Distance(pa.cfg.Subnet.IP, allocations[0].StartIP) > 2 {
```

Comment:
why did you drop calls to ip.Network() ?

a user may provide a subnet like: 192.168.100.44/24

Reply:
ParseCIDR returns two objects - net.IP and net.IPNet. We use the second one. It contains valid network ip.